pkg/policy/resolve_test.go

```diff
@@ -51,23 +51,17 @@ var (
 type dummyEndpoint struct {
 	ID               uint16
 	SecurityIdentity *identity.Identity
+	Endpoint         // Implement methods of the interface that need to mock out real behavior.
```

Comment:
I think this only declares that `DummyEndpoint` implements these methods, it does not actually provide any implementations of them. This is what's causing the panic in the Travis build:

```
[signal SIGSEGV: segmentation violation code=0x1 addr=0x38 pc=0x1a17733]
goroutine 1117 [running]:
github.com/cilium/cilium/pkg/policy.(*dummyEndpoint).IsHost(0xc000ff6d80, 0x2e3dd60)
	<autogenerated>:1 +0x33
github.com/cilium/cilium/pkg/policy.ruleSlice.updateEndpointsCaches(0xc000fc0000, 0x3e8, 0x3e8, 0x21061e0, 0xc000ff6d80, 0x0, 0x0, 0x0)
	/home/travis/gopath/src/github.com/cilium/cilium/pkg/policy/rules.go:165 +0x123
github.com/cilium/cilium/pkg/policy.ruleSlice.UpdateRulesEndpointsCaches.func1(0x21061e0, 0xc000ff6d80)
	/home/travis/gopath/src/github.com/cilium/cilium/pkg/policy/repository.go:446 +0x85
github.com/cilium/cilium/pkg/policy.(*EndpointSet).ForEachGo.func1(0xc0007640f0, 0xc000a1ead0, 0x21061e0, 0xc000ff6d80)
	/home/travis/gopath/src/github.com/cilium/cilium/pkg/policy/identifier.go:71 +0x42
created by github.com/cilium/cilium/pkg/policy.(*EndpointSet).ForEachGo
	/home/travis/gopath/src/github.com/cilium/cilium/pkg/policy/identifier.go:70 +0x13f
```

In this case, there is no actual implementation of `IsHost` for `dummyEndpoint`.

Reply:
Alternatively, when the `dummyEndpoint` is created, its `Endpoint` value needs to be set to something that includes an  implementation of `IsHost`.

Reply:
Thanks for the review. Yes, the main motivation here is that users are not forced to implement all the method of an interface, rather only the ones that are relevant to the unit test cases. 
I'll triage the Travis failure (looks like the test passed in my local environment). Thanks!

Reply:
As it turns out, `IsHost()` method was needed in one of the test permutations that Travis does. 

>Endpoint         // Implement methods of the interface that need to mock out real behavior.
Based on the comment, implementing the method.